sources/sql_database/schema_types.py

```diff
@@ -158,4 +211,4 @@ def row_tuples_to_arrow(
             )
             float_array = pa.array(columnar[field.name], type=pa.float64())
             columnar[field.name] = float_array.cast(field.type, safe=False)
-    return pa.Table.from_pydict(columnar, schema=arrow_schema)
```

Comment:
why this is removed? here I'm afraid all the precision information etc. will be lost because pyarrow will infer schema from the data and we know way more from sql schema.

ofc we have a problem with unknown data types. but we can create a table for those columns separately and then join it into one. WDYT? should be cheap, arrow is columnar

Reply:
Ah mistake there. Didn't mean to keep this change in :grimacing: